modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -308,11 +312,17 @@ PinMeshGenerator::PinMeshGenerator(const InputParameters & parameters)
         // and no ring regions
         else if (!_quad_center)
         {
-          background_blk_ids[0] = pin_block_id_tri;
+          background_blk_ids[0] = RGMB::PIN_BLOCK_ID_TRI;
           background_blk_names[0] += "_TRI";
         }
       }
 
+      // If flexible assembly stitching is invoked and this is an assembly mesh with only a
+      // background region, do not call mesh subgenerators here This assembly mesh should be created
+      // entirely in generateFlexibleAssemblyBoundaries()
+      bool skip_assembly_generation =
+          _is_assembly && use_flexible_stitching && _intervals.size() == 1;
+
       // Generate Cartesian/hex pin using PolygonConcentricCircleMeshGenerator
       {
```

Comment:
could you have if(skip_assembly_generation) here instead? 
Makes more sense than almost creating the generators then not adding them

Reply:
Done